ISLANDORA-2053: Update travis.yml to force PHP 5.3.3 to run under Ubuntu Precise

DiegoPino opened the pull request:
**ISLANDORA-2053**: (https://jira.duraspace.org/browse/ISLANDORA-2053)
http://irclogs.islandora.ca/2017-09-05.html
https://github.com/travis-ci/travis-ci/issues/2963
https://github.com/Islandora/islandora/pull/683

As of September 2017, Travis-CI has removed support for PHP 5.3 (and 5.2) for their Ubuntu Trusty Images, forcing us, in order  to keep Drupal's compatibility matrix tested, to run Travis-CI test for 5.3 on Ubuntu Precise

# What does this Pull Request do?

This pull Fixes our failing Travis-CI tests: 
Should allow PHP 5.3.3 to run under Ubuntu Precise and Higher versions under Ubuntu Trusty.
Changes the way we define which PHP versions and Fedora ones run in Travis

# What's new?
We moved PHP 5.3.3 version and the corresponding Fedora versions into the matrix section of the Travis YAML file. 

# How should this be tested?
 Travis-CI should pass and everything single PHP/FEDORA version combination should look good and very green.

# Interested parties
@Islandora/7-x-1-x-committers

DiegoPino committed:
force PHP 5.3.3 to run under Ubuntu Precise

whikloj commented:
@DiegoPino `dist: trusty` has changed package names for tesseract too. We are going to need some more complex logic for installing these different packages I think.

DiegoPino commented:
@whikloj 😢 But most of this stuff actually runs on vagrant which is trusty... ok, the fun never ends!

DiegoPino committed:
Check for php 5.3.3 == Precise

If Precise do it like before, if trusty (any other PHP) apt-get
Simpler than creating a new ENV var

DiegoPino committed:
No need to remove the temp directory

Not sure why we had that, travis images are wiped at the end

whikloj commented on tests/scripts/tesseract.sh:
```diff
@@ -1,15 +1,19 @@
+  tar xf tessdata.tar.gz && rm -rf tessdata.tar.gz
+  cp tessdata/* /usr/local/share/tessdata/
+  sudo ldconfig
+  echo -e "\ntesseract output:"
```
I think this was meant as context for the `tesseract --version && tesseract --list-langs && cd ~ && rm -rf ~/tesseract` line, so perhaps moving it after the `if ... else .. fi` makes sense.

DiegoPino replied:
Right! You are the best 👍 Doing it now

whikloj commented on tests/scripts/tesseract.sh:
```diff
@@ -1,15 +1,19 @@
+fi
 
-tesseract --version && tesseract --list-langs && cd ~ && rm -rf ~/tesseract
+tesseract --version && tesseract --list-langs
```
Should you move the `cd ~ && rm -rf ~/tesseract` up into the end of the `if [[ $TRAVIS_PHP_VERSION == '5.3.3' ]]; then` block?

whikloj replied:
Actually not sure why we need to `cd ~` and then `rm ~/tesseract` seems like just the `rm` would work.

DiegoPino replied:
I think we don't even need that. Travis images get wiped out after they finish so i removed it completely because it seemed to me like a waste of code? or time? or cpu? or energy?

DiegoPino committed:
Moving output into context

whikloj commented:
Looks like there are a variety of tests that are now failing. https://travis-ci.org/Islandora/islandora_solution_pack_book/jobs/272978061#L1017

DiegoPino commented:
@whikloj oh gosh. totally 😭 And the environmental vars are not fetched by precise? WHY! Anyway, will take some time to debug again, the OCR test that are failing are probably related to Tesseract version...

DiegoPino committed:
Copy cat to newspaper

Was known for working correctly

DiegoPino commented:
@whikloj @rosiel ok, got the issue (i hate this sometimes)
Tesseract that Travis fetches for Ubuntu Trusty is 3.03. That one detects the multiple "TIFF" words inside the test case image file we are using as "T|FF" and the tests fail.

@jonathangreen spoke about unifying the way we deploy Tesseract via a York hosted binary, but the dependencies are hard to maintain. I agree on a common approach (i would go for apt-get install simply) but i also think it would be kinda simpler to change the test case, for one that actually works out of the box with an ```I``` that does not look like an ```|```. Not our fault that the language training files differ. DO i get a YEAH! go for it?

DiegoPino committed:
Lets see if i got tests right

Tesseract 3.03 thinks that our previous TIFF TIFF TIFF are actually
T|FF T|FF, so instead of compiling a new tesseract for Nx6 options,
lets change the tests

rosiel committed:
Fix tests

DiegoPino commented:
@rosie. did i get the happy dogs and cats wrong?

DiegoPino commented:
I did! Thanks for correcting

rosiel reviewed (approved):
Looks good! (Forgive me, I fixed the numbers of happy cats and dogs so that it would pass - now that I've committed to this PR I don't think I can merge, but I think I can review?)

DiegoPino commented:
How did you changed that? I was doing it from my diego branch so... 🙂Using
your push priv? Well you can't merge so will wait for @whikloj  or @jonathangreen 
Normally what we do here is you do a pull request against my branch, i
merge into mine etc but co release manager priv go over that. No rush now
since it is passing!


-- 
Diego Pino Navarro
Digital Repositories Developer
Metropolitan New York Library Council (METRO)

jonathangreen commented:
Since we changed the tests, is there anything stopping us from just apt-get installing tesseract instead of compiling it?

DiegoPino commented:
@jonathangreen eh.. not sure. Do you want me to test? the compile is just for precise (so only a fraction of the total). Let me do later tonight a try and lets see. Please don´t pull against my pull since that will block you also out! thanks =)

DiegoPino committed:
test changed. we can try with apt-get tesseract

If this passes, then it could stick.

DiegoPino commented:
@jonathangreen yep 🎉  It passes: https://travis-ci.org/Islandora/islandora_solution_pack_book/jobs/274422828#L807!!

So. There is no harm in letting the "compile" part which in the rare event that apt-get install fails, which should never happen.I  can for sure simply clear out all that extra script and leave the simple two line statement. You all decide? Here is the diff.

https://github.com/Islandora/islandora_solution_pack_book/pull/198/files#diff-80dc3788fff4c781859827ef53c8dee9

Thanks!

jonathangreen commented:
@DiegoPino I'm happy with it as is, but it looks like its failing on PHP 5.3.3 :(

DiegoPino commented:
@jonathangreen ahhhhhh ahhhhh ahhh. Ok, got it. So tesseracts version output has a different exit code than 3.03... i wish we could maintain only PHP 7.1 and 16.04.. 😢

DiegoPino committed:
We can't be sure --list-langs will work on precise

that. So that, just show the version.. this is just test, do we need to
show the languages if it won’t work?

DiegoPino commented:
Not sure what the heck is happening now, HOCR Tests don´t pass.. failing in 5.3.3 ? Before, when compiling (a few pulls back) travis was green for 5.3.3.. maybe just too too late. 😢 should i keep persisting in this?  maybe HOCR needs different parameters for tesseract 3.02? I could go back to where all was green...

DiegoPino committed:
Rolling back, no HOCR with 3.02

rosiel commented:
@diegopino I am not sure how but on github I edited the file from the PR and it offered to let me push directly to your issue branch. Either it's unprotected or I have goddess-like github powers. Any way I realize I shouldn't have done that and will PR in the future.

DiegoPino commented:
@rosiel, not unprotected for sure.! You used your push privileges as co manager to modify the live pull. Please don't do it again, thanks!
